coral-hive/src/main/java/com/linkedin/coral/transformers/CoralRelNodeToCoralSqlNodeTransformer.java

```diff
@@ -0,0 +1,172 @@
+
+  /**
+   * Creates a CoralRelNodeToCoralSqlNodeTransformer.
+   */
+  CoralRelNodeToCoralSqlNodeTransformer() {
+    super(INSTANCE);
+  }
+
+  private static SqlDialect returnInstance() {
+    SqlDialect.Context context = SqlDialect.EMPTY_CONTEXT.withDatabaseProduct(SqlDialect.DatabaseProduct.HIVE)
+        .withNullCollation(NullCollation.HIGH);
+
+    return new SqlDialect(context);
+  }
+
+  /**
+   * This overridden function makes sure that the basetable names in the output SqlNode
+   * will be in the format - "dbname.tablename" instead of "catalogname.dbname.tablename"
+   * For Example:
+   * hive.default.complex will be converted to default.complex
+   *
+   * input RelNode format is:
+   * LogicalTableScan(table=[[hive, default, complex]])
+   * This overriding function creates a SqlIdentifier(default, complex)
+   * instead of super's default: SqlIdentifier(hive, default, complex)
+   */
+  @Override
+  public Result visit(TableScan e) {
+    //todo: look into removing this transformation during hiveToRelConversion
+    checkQualifiedName(e.getTable().getQualifiedName());
+    List<String> tableNameWithoutCatalogAndDb = e.getTable().getQualifiedName().subList(1, 3);
+    final SqlIdentifier identifier = new SqlIdentifier(tableNameWithoutCatalogAndDb, SqlParserPos.ZERO);
+    return result(identifier, ImmutableList.of(Clause.FROM), e, null);
+  }
+
+  /**
+   * Correlate represents the FROM clause of the final sql query
+   * when the FROM clause contains explode() UDF as well
+   *
+   * Overriding the default implementation is required because
+   * the rightChildResult is also overriden and doesn't match super's expectation
+   */
+  @Override
+  public Result visit(Correlate e) {
+    //Left Result removes the catalog name "hive" from the table path
+    final Result leftResult = visitChild(0, e.getLeft());
+
+    // Add context specifying correlationId has same context as its left child
+    correlTableMap.put(e.getCorrelationId(), leftResult.qualifiedContext());
+
+    final Result rightResult = visitChild(1, e.getRight());
+
+    SqlJoin join = new SqlJoin(POS, leftResult.asFrom(), SqlLiteral.createBoolean(false, POS),
+        JoinType.COMMA.symbol(POS), rightResult.asFrom(), JoinConditionType.NONE.symbol(POS), null);
+
+    SqlIdentifier leftNode = (SqlIdentifier) leftResult.node;
+    SqlBasicCall lateralRightNode = (SqlBasicCall) rightResult.node;
+    SqlBasicCall asSqlNode = (SqlBasicCall) lateralRightNode.getOperands()[0];
+    SqlIdentifier asSqlNodeAlias = (SqlIdentifier) asSqlNode.getOperands()[1];
+
+    //The default implementation assumes that the rightLateralNode, for LateralSqlOperator,
+    // has a null alias. At the same time, it expects a non-null alias for the same node.
+    // At this point, we can either customize the rightLateralNode to pass the alias check
+    // or we can bypass the check and work around it. The latter appoach has been chosen.
```

Comment:
Could you elaborate `bypass the check and work around it`? Preferably add an example in the comment.

Reply:
marking as resolved since this code change was dropped.